src/Inciter/AMR/node_store.h

```diff
@@ -0,0 +1,245 @@
+#ifndef AMR_node_store_h
+#define AMR_node_store_h
+
+#include <cmath>
+
+#include "types.h"
+#include "tet_store.h"
+
+// TODO: make this have a base class to support multiple generator schemes
+// using the policy design pattern
+namespace AMR {
+
+    class node_store_t
+    {
+
+        private:
+            coord_type* m_x;
+            coord_type* m_y;
+            coord_type* m_z;
+
+            // We really don't want people to pass this by value..
+            // (Because we store refs in here, which are consts..)
+            // NonCopyable & operator=(const NonCopyable&) = delete;
+            node_store_t(const node_store_t& c) = delete;
+            node_store_t& operator=(const node_store_t&) = delete;
+
+        public:
+            node_store_t() { } // default cons
```

Comment:
... and the default constructor leaving the state pointers uninitialized: a recipe for disaster. ;-)